Assets/Scripts/DialogueBubble.cs

```diff
@@ -0,0 +1,275 @@
+                int vertexIndex = textInfo.characterInfo[i].vertexIndex;
+
+                // Get the cached vertices of the mesh used by this text element (character or sprite).
+                Vector3[] sourceVertices = cachedMeshInfo[materialIndex].vertices;
+
+                // Determine the center point of each character at the baseline.
+                //Vector2 charMidBasline = new Vector2((sourceVertices[vertexIndex + 0].x + sourceVertices[vertexIndex + 2].x) / 2, charInfo.baseLine);
+                // Determine the center point of each character.
+                Vector2 charMidBasline = (sourceVertices[vertexIndex + 0] + sourceVertices[vertexIndex + 2]) / 2;
+
+                // Need to translate all 4 vertices of each quad to aligned with middle of character / baseline.
+                // This is needed so the matrix TRS is applied at the origin for each character.
+                Vector3 offset = charMidBasline;
+
+                Vector3[] destinationVertices = textInfo.meshInfo[materialIndex].vertices;
+
+                destinationVertices[vertexIndex + 0] = sourceVertices[vertexIndex + 0] - offset;
+                destinationVertices[vertexIndex + 1] = sourceVertices[vertexIndex + 1] - offset;
+                destinationVertices[vertexIndex + 2] = sourceVertices[vertexIndex + 2] - offset;
+                destinationVertices[vertexIndex + 3] = sourceVertices[vertexIndex + 3] - offset;
+
+                vertAnim.angle = Mathf.SmoothStep(-vertAnim.angleRange, vertAnim.angleRange, Mathf.PingPong(loopCount / 25f * vertAnim.speed, 1f));
+                Vector3 jitterOffset = new Vector3(Random.Range(-.25f, .25f), Random.Range(-.25f, .25f), 0);
+
+                matrix = Matrix4x4.TRS(jitterOffset * CurveScale, Quaternion.Euler(0, 0, Random.Range(-5f, 5f) * AngleMultiplier), Vector3.one);
+
+                destinationVertices[vertexIndex + 0] = matrix.MultiplyPoint3x4(destinationVertices[vertexIndex + 0]);
+                destinationVertices[vertexIndex + 1] = matrix.MultiplyPoint3x4(destinationVertices[vertexIndex + 1]);
+                destinationVertices[vertexIndex + 2] = matrix.MultiplyPoint3x4(destinationVertices[vertexIndex + 2]);
+                destinationVertices[vertexIndex + 3] = matrix.MultiplyPoint3x4(destinationVertices[vertexIndex + 3]);
+
+                destinationVertices[vertexIndex + 0] += offset;
+                destinationVertices[vertexIndex + 1] += offset;
+                destinationVertices[vertexIndex + 2] += offset;
+                destinationVertices[vertexIndex + 3] += offset;
+
+                vertexAnim[i] = vertAnim;
+            }
+
+            // Push changes into meshes
+            for (int i = 0; i < textInfo.meshInfo.Length; i++)
+            {
+                textInfo.meshInfo[i].mesh.vertices = textInfo.meshInfo[i].vertices;
+                Text.UpdateGeometry(textInfo.meshInfo[i].mesh, i);
+            }
+
+            loopCount += 1;
```

Comment:
how often will this update? As an int, is it possible to go out of bounds?

Reply:
`loopCount` is used in this function, it needs to be of incrementing value

https://docs.unity3d.com/ScriptReference/Mathf.PingPong.html

Reply:
I know it's used in the function, that's all good. I'm just wondering how many times this big while loop is iterated over. If it goes over Integer.MaxValue, maybe there would be a problem. Does it ever reset?

Reply:
Anyway, all good for now!

Reply:
Increments ~200 per second would take almost 3000 hrs to overflow